lxd/devlxd_instances.go

```diff
@@ -3,10 +3,29 @@ package main
 import (
 	"github.com/canonical/lxd/lxd/device/filters"
 	"github.com/canonical/lxd/lxd/instance"
+	"github.com/canonical/lxd/shared/api"
 )
 
 type devLXDDeviceAccessValidator func(device map[string]string) bool
 
+// getOwnedDevices extracts instance devices that are owned by the provided identity.
+// Two maps are returned, one containing owned devices, and the other containing the remaining
+// devices.
+func getOwnedDevices(inst api.Instance, identityID string) (ownedDevices map[string]map[string]string, otherDevices map[string]map[string]string) {
```

Comment:
remember you are in the `main` package and so IDEs will suggest functions to use for the whole of LXD.

So make sure that if a function or variable in the main package is devlxd specific (as this one is) that its prefixed with `devLXD`.